src/gluonts/mx/distribution/lds.py

```diff
@@ -211,33 +213,105 @@ def log_prob(
-
-            # Sum `ladj` over all time steps.
-            log_p = F.broadcast_add(log_p, ladj)
+        F = self.F
+        x_transformed, ladj = self.transform(x, scale, output_transform)
+        log_p, final_mean, final_cov = self.kalman_filter(
+            x_transformed, observed
+        )
+
+        if scale is not None or output_transform is not None:
+            # Get the shape right if only scaling is done.
+            ladj = F.broadcast_add(ladj, F.zeros_like(log_p))
+            # Mask `ladj` for missing values as these should not modify log prob.
+            ladj = F.where(
+                condition=observed,
+                x=ladj * observed,
+                y=self.F.zeros_like(ladj),
+            )
+            log_p = log_p + ladj
+
         return log_p, final_mean, final_cov
 
+    def transform(
+        self,
+        x: Tensor,
+        scale: Optional[Tensor] = None,
+        output_transform: Optional[ComposedBijectionHybridBlock] = None,
+    ):
+        """
+        Scales and then transforms the target `x` by the (inverse of) `output_transform`.
+
+        Parameters
+        ----------
+        x
+            Observations, shape (batch_size, seq_length, output_dim)
+        scale
+            Scale of each sequence in x, shape (batch_size, output_dim)
+        output_transform
+            Specifies the (inverse of) transformation to be applied to the target time series.
+            Given target `x` is first scaled then the inverse of `output_transform` is applied to make it more Gaussian-like.
+
+        Returns
+        -------
+        Tensor
+            Transformed target, shape (batch_size, seq_length, output_dim)
+            log_abs_det_jac,    shape (batch_size, 1) if `output_transform` is None, (batch_size, seq_length) otherwise
+        """
+        F = self.F
+
+        def _scale(x):
+            x_scaled, ladj = x, None
+            if scale is not None:
+                # We apply scaling before transforming the target by the given output transformation.
+                x_scaled = F.broadcast_div(x, scale.expand_dims(axis=1))
+
+                # log_abs_det_jac: sum over all output dimensions. Shape (batch_size, 1)
+                ladj = -F.sum(F.log(F.abs(scale)), axis=-1, keepdims=True)
+            return x_scaled, ladj
+
+        def _output_transform(x):
+            x_transformed, ladj = x, None
+            if output_transform is not None:
+                x_transformed = output_transform.f_inv(x)
+
+                # We need the `ladj` of the inverse transform but we compute `ladj` of the forward transform which only
+                # differs by sign. Shape (batch_size, seq_length)
+                ladj = -output_transform.log_abs_det_jac(x, x_transformed)
+            return x_transformed, ladj
+
+        x_scaled, ladj_scaling = _scale(x)
+        x_scaled_transformed, ladj_transform = _output_transform(x_scaled)
+
+        if scale is not None and output_transform is not None:
```

Comment:
Minor comment:  Here you could return `ladj_scaling` and  `ladj_transform` as `F.zeros_like()` instead of `None` so you can add them. This would make the if else more compact to something like
```
ladj = F.broadcast_add(ladj_scaling, ladj_transform) if scale is not None or output_transform is not None else None
```
 I don't have a strong opinion on what is the best approach though.